image/signature/policy_eval_signedby.go

```diff
@@ -40,9 +43,58 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(ctx context.Context, image priva
 	}
 
 	// FIXME: move this to per-context initialization
-	mech, trustedIdentities, err := newEphemeralGPGSigningMechanism(data)
-	if err != nil {
-		return sarRejected, nil, err
+	// Import the keys with a 60s timeout to avoid hanging indefinitely. see issues.redhat.com/browse/OCPBUGS-57893
```

Comment:
I really quite strongly don’t want this complexity in core policy enforcement code.

Also, I think the core complexity of the cleanup attempt here comes from separating the `new…Mechanism` initialization and `defer mech.Close()` into different goroutines. I think running _all_ of the `PolicyContext.IsRunningImageAllowed` within a goroutine with a timeout would automatically handle cleanup (… or not, and leak a C thread+goroutine). In what respect is this approach better?

---

This is really orthogonal to the other part of the of the PR: _if_ the approach to terminate GPGME via a callback failing works, this moving of the code into a concurrent goroutine does not need to exist (and continues to have the issues of #421 ).